playgame.py

```diff
@@ -5,55 +5,111 @@
 from tools.hanabi_card import HanabiColor
 import sys
 import logging
+from logging.handlers import RotatingFileHandler
+import os
 from tools.hanabi_moves import \
     HanabiDiscardAction, \
     HanabiPlayAction, \
     HanabiColorDiscloseAction, \
     HanabiRankDiscloseAction
 
-logger = logging.getLogger(__name__)
-
 def main(argv):
     args = parse_args()
-    prep_logger(args.log_dir, args.verbose, args.log_stderr)
+    logger = prep_logger(args.log_dir, args.verbose, args.log_stderr, len(args.players))
+    if args.is_tournament:
+        run_tournament(args, logger)
+    else:
+        run_one_game(args, logger)
+
+def generate_pairings(players):
+    res = []
+    for i in range(len(players)):
+        for j in range(i + 1, len(players)):
+            res.append([players[i], players[j]])
+    return res
+
+def run_tournament(args, logger):
+    tournament_scores = dict.fromkeys(args.players, 0)
+    pairings = generate_pairings(args.players)
+    for player1, player2 in pairings:
+        try:
+            game = HanabiGame([player1, player2], args.seed, args.variant)
+            game.play_game(args, logger)
+            score = game.table.score()
+        except:
+            #if a player has messed up, penalize both
+            #could diqualify the failing bot if we can determine who failed
+            logger.warning("{player1} or {player2} failed to complete a game".format(
+                player1 = player1,
+                player2 = player2
+                ))
+            score = -25
+        finally:
+            for handler in logger.handlers:
+                if handler.__class__ is RotatingFileHandler:
+                    handler.doRollover()
+            tournament_scores[player1] += score
+            tournament_scores[player2] += score
+    tournament_scores = {k: v / len(pairings) for k, v in tournament_scores.iteritems()}
+    logger.info("Scores: {scores}".format(scores = tournament_scores))
```

Comment:
Should these messages use single quotes to match the others?

The last time I looked, linting options in Python were pretty limited, but this seems like something it would be nice to have a linter to check (and possibly auto-fix).

Reply:
yup. I've been trying soooo hard to fix these